File: 2685.py
from collections import deque

class Solution:
    def countCompleteComponents(self, n: int, edges: list[list[int]]) -> int:
        graph = [[] for _ in range(n)]
        for u, v in edges:
            graph[u].append(v)
            graph[v].append(u)

        visited = [False] * n
        complete_count = 0

        for node in range(n):
            if not visited[node]:
                visited[node] = True
                stack = [node]
                node_count = 0
                edge_count = 0

                while stack:
                    curr = stack.pop()
                    node_count += 1
                    edge_count += len(graph[curr])

                    for neighbor in graph[curr]:
                        if not visited[neighbor]:
                            visited[neighbor] = True
                            stack.append(neighbor)

                if edge_count // 2 == node_count * (node_count - 1) // 2:
                    complete_count += 1

        return complete_count

File: 2685.js
/**
 * @param {number} n
 * @param {number[][]} edges
 * @return {number}
 */
var countCompleteComponents = function(n, edges) {
    const graph = Array.from({ length: n }, () => []);
    for (const [u, v] of edges) {
        graph[u].push(v);
        graph[v].push(u);
    }

    const visited = new Array(n).fill(false);
    let count = 0;

    for (let node = 0; node < n; node++) {
        if (!visited[node]) {
            visited[node] = true;
            let edgeCount = 0;
            let nodeCount = 0;
            const stack = [node];

            while (stack.length) {
                const curr = stack.pop();
                edgeCount += graph[curr].length;
                nodeCount += 1;

                for (const next of graph[curr]) {
                    if (!visited[next]) {
                        visited[next] = true;
                        stack.push(next);
                    }
                }
            }

            if (edgeCount / 2 === (nodeCount * (nodeCount - 1)) / 2) {
                count += 1;
            }
        }
    }

    return count;
};
